src/uct/sm/knem/knem_iface.c

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) UT-Battelle, LLC. 2014-2015. ALL RIGHTS RESERVED.
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+ * $COPYRIGHT$
+ * $HEADER$
+ */
+
+#include "knem_pd.h"
+#include "knem_iface.h"
+#include "knem_ep.h"
+
+#include <uct/api/addr.h>
+#include <uct/tl/context.h>
+
+UCT_PD_REGISTER_TL(&uct_knem_pd_component, &uct_knem_tl);
+
+static ucs_config_field_t uct_knem_iface_config_table[] = {
+    {"", "ALLOC=huge,mmap,heap", NULL,
+    ucs_offsetof(uct_knem_iface_config_t, super),
+    UCS_CONFIG_TYPE_TABLE(uct_iface_config_table)},
+    {NULL}
+};
+
+static ucs_status_t uct_knem_iface_get_address(uct_iface_t *tl_iface,
+                                             struct sockaddr *addr)
+{
+    uct_sockaddr_process_t *iface_addr = (void*)addr;
+    iface_addr->sp_family = UCT_AF_PROCESS;
+    iface_addr->node_guid = ucs_machine_guid();
+    return UCS_OK;
+}
+
+static int uct_knem_iface_is_reachable(uct_iface_t *tl_iface,
+                                     const struct sockaddr *addr)
+{
+    return (addr->sa_family == UCT_AF_PROCESS) &&
+           (((uct_sockaddr_process_t*)addr)->node_guid == ucs_machine_guid());
+}
+
+static ucs_status_t uct_knem_flush()
+{
+    return UCS_OK;
+}
+
+static ucs_status_t uct_knem_iface_query(uct_iface_h tl_iface,
+                                       uct_iface_attr_t *iface_attr)
+{
+    memset(iface_attr, 0, sizeof(uct_iface_attr_t));
+
+    /* default values for all shared memory transports */
+    iface_attr->cap.put.max_zcopy      = 64*1024*1024; /* Pasha: tmp value */
```

Comment:
what fails exactly? knem copy? or memory allocation? if it is the latter, the correct place to fix is the tests